dora/core/common/src/main/java/alluxio/util/io/FileUtils.java

```diff
@@ -98,6 +102,67 @@ public static void changeLocalFilePermission(String filePath, String perms) {
     }
   }
 
+  /**
+   * Changes local file's user defined file attribute.
+   * @param filePath the file to change attribute
+   * @param name the name of attribute
+   * @param value the value of attribute
+   */
+  public static void changeLocalFileUserDefinedAttribute(String filePath, String name,
+                                                         byte[] value) {
+    try {
+      UserDefinedFileAttributeView attributeView =
+          Files.getFileAttributeView(Paths.get(filePath), UserDefinedFileAttributeView.class);
+      if (attributeView == null) {
+        throw new UnimplementedRuntimeException("set attribute is not implemented");
+      }
+      attributeView.write(name, ByteBuffer.wrap(value));
+    } catch (UnsupportedOperationException e) {
+      throw new UnimplementedRuntimeException(e, ErrorType.External);
+    } catch (ClassCastException | IllegalArgumentException e) {
+      throw new InvalidArgumentRuntimeException(e);
+    } catch (SecurityException e) {
+      throw new PermissionDeniedRuntimeException(e);
+    } catch (IOException e) {
+      throw new UnknownRuntimeException(e);
+    }
+  }
+
+  /**
+   * Gets local file's user defined file attribute.
+   *
+   * @param filePath
+   * @return the map of attribute view
+   */
+  public static Map<String, String> getLocalFileUserDefinedAttribute(String filePath) {
+    try {
+      UserDefinedFileAttributeView attributeView =
+          Files.getFileAttributeView(Paths.get(filePath), UserDefinedFileAttributeView.class);
+      if (attributeView == null) {
+        throw new UnimplementedRuntimeException("set attribute is not implemented");
+      }
+      Map<String, String> attrMap = new HashMap<>();
+      for (String attributeName : attributeView.list()) {
+        int attributeSize = attributeView.size(attributeName);
+        ByteBuffer attributeBuffer = ByteBuffer.allocate(attributeSize);
```

Comment:
sanity check ^

Reply:
I used `ByteBuffer.allocate` here instead of `ByteBuffer.allocateDirect`, so the buffer is managed by JVM. it will be automatically destroyed and recycled after GC.